### PR TITLE
feat(authorizenet): add driver errors

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: libs/authorizenet/driver/src/errors/auth-failed.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { DaffError, DaffInheritableError } from '@daffodil/core';
+
+/**
+ * An error thrown when either the API Login ID or Public Client Key values are incorrect.
+ */
+export class DaffAuthorizeNetAuthFailedError extends DaffInheritableError implements DaffError {
```
Comment (lderrickable) [inline libs/authorizenet/driver/src/errors/auth-failed.ts:6]: What's the difference between this error and `DaffAuthorizeNetInvalidClientKeyError` and `DaffAuthorizeNetInvalidLoginIdError`?

Comment (griest024) [reply libs/authorizenet/driver/src/errors/auth-failed.ts:6]: To be honest, I'm not completely sure. This is one of the ones that might be able to be condensed. I'm referencing [this page](https://developer.authorize.net/api/reference/features/acceptjs.html) for the error codes and there are 3 distinct errors.

I think the main difference is that the other errors will be thrown for missing keys whereas this one is just for invalid ones. I'm not totally sure about that however.

Comment (lderrickable) [reply libs/authorizenet/driver/src/errors/auth-failed.ts:6]: oh ok, yea that sounds fine then

**Review by damienwebdev — APPROVED** [review]
lgtm

**Pull request merged**